models/fulfilmentConfirmed.go

```diff
@@ -0,0 +1,40 @@
+package models
+
+import (
+	"errors"
+	"fmt"
+	"github.com/ONSdigital/census-rm-pubsub-adapter/validate"
+)
+
+type FulfilmentConfirmed struct {
+	TimeCreated     *HazyUtcTime `json:"dateTime" validate:"required"`
+	TransactionId   string       `json:"transactionId" validate:"required"`
+	ProductCode     string       `json:"productCode" validate:"required"`
+	Channel         string       `json:"channel" validate:"required"`
+	QuestionnaireId string       `json:"questionnaireId"`
+	CaseRef         string       `json:"caseRef"`
+}
+
+func (o FulfilmentConfirmed) GetTransactionId() string {
+	return o.TransactionId
+}
+
+func (o FulfilmentConfirmed) Validate() error {
+	err := validate.Validate.Struct(o)
+
+	if err == nil {
+		if o.Channel == "QM" {
+			if len(o.QuestionnaireId) == 0 {
+				err = errors.New(fmt.Sprintf("Missing questionnaire ID in QM message: %T, tx_id: %q", o, o.GetTransactionId()))
+			}
+		} else if o.Channel == "PPO" {
+			if len(o.CaseRef) == 0 {
```

Comment:
nested ifs could be combined

Reply:
it's the same amount of code, but whatever